### PR TITLE
Ravinder | Staging | GIDK-693 | Purchase Order

### PR DESCRIPTION
**Pull request by ravinderwalkover**
Ravinder | Staging | GIDK-693 | Purchase Order

### CODE REVIEW

File: apps/web-giddh/src/app/invoice/invoice.component.html [hunk 1]
```diff
@@ -144,12 +144,39 @@
                                     <app-invoice-setting></app-invoice-setting>
                                 </div>
                             </tab>
+                            <tab heading="Purchase Order">
+                                <div>
+                                    <!-- <purchase-order></purchase-order> -->
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/invoice/invoice.component.html:149]: Tab with empty!  
why commented?

File: apps/web-giddh/src/app/invoice/invoice.component.html [hunk 2]
```diff
@@ -144,12 +144,39 @@
...
+                                </div>
+                            </tab>
+                            <tab heading="Purchase Bill">
+                                <div>
+                                    <!-- <app-invoice-preview></app-invoice-preview> -->
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/invoice/invoice.component.html:154]: Tab with empty!
why commented?

File: apps/web-giddh/src/app/invoice/preview/models/purchase-bill-template-preview/purchase-bill-template-preview.component.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+@Component({
+    selector: 'purchase-bill-template',
+    templateUrl: './purchase-bill-template-preview.component.html',
+    styleUrls: ['./purchase-bill-template-preview.component.scss']
+})
+export class PurchaseBillTemplateComponent implements OnInit {
+    public modelRef: BsModalRef;
+    public modalRef: BsModalRef;
+    public isMulticurrencyAccount: true;
+    public orderHistoryAsideState: string = 'out';
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/invoice/preview/models/purchase-bill-template-preview/purchase-bill-template-preview.component.ts:21]: Missing JS doc for all variables

File: apps/web-giddh/src/app/purchase/create-purchase-order/create-purchase-order.component.ts [hunk 4]
```diff
@@ -0,0 +1,2717 @@
...
+     */
+    public startLoader(shouldStartLoader: boolean): void {
+        if (this.showLoaderUntilDataPrefilled) {
+            this.showLoader = true;
+        } else {
+            this.showLoader = shouldStartLoader;
+        }
+    }
+
+    private modifyStateResp(stateList: StateCode[]): IOption[] {
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/purchase/create-purchase-order/create-purchase-order.component.ts:878]: Add JS doc

File: apps/web-giddh/src/app/purchase/create-purchase-order/create-purchase-order.component.ts [hunk 5]
```diff
@@ -0,0 +1,2717 @@
...
+     *
+     * @private
+     * @param {WarehouseDetails} [warehouse] Warehouse to show pre-filled in drop down
+     * @memberof CreatePurchaseOrderComponent
+     */
+    private initializeWarehouse(warehouse?: WarehouseDetails): void {
+        this.store.pipe(select(appState => appState.warehouse.warehouses), filter((warehouses) => !!warehouses), take(1)).subscribe((warehouses: any) => {
+            if (warehouses) {
+                const warehouseData = this.settingsUtilityService.getFormattedWarehouseData(warehouses.results);
+                this.warehouses = warehouseData.formattedWarehouses;
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/purchase/create-purchase-order/create-purchase-order.component.ts:897]: Add nullity check for warehouseData

File: apps/web-giddh/src/app/purchase/create-purchase-order/create-purchase-order.component.ts [hunk 6]
```diff
@@ -0,0 +1,2717 @@
...
+                let description = this.description.toArray();
+                if (description && description[this.activeIndex]) {
+                    description[this.activeIndex].nativeElement.focus();
+                }
+            }, 200);
+            return txn;
+        }
+    }
+
+    public onNoResultsClicked(index: any): void {
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/purchase/create-purchase-order/create-purchase-order.component.ts:1307]: Remove if not required

File: apps/web-giddh/src/app/purchase/order-history/order-history.component.ts [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    public  isMulticurrencyAccount : true;
+    @Output() public closeAsideEvent: EventEmitter<boolean> = new EventEmitter(true);
+    constructor(
+        private modalService: BsModalService,
+        private generalService: GeneralService,
+    ) { }
+    ngOnInit() {
+    }
+   
+    public closeAsidePane(event) {
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/purchase/order-history/order-history.component.ts:29]: Add JS doc and return type

File: apps/web-giddh/src/app/purchase/order-history/order-history.component.ts [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+import { GeneralService } from 'apps/web-giddh/src/app/services/general.service';
+@Component({
+    selector: 'aside-order-history',
+    templateUrl: './order-history.component.html',
+    styleUrls: ['./order-history.component.scss']
+})
+export class OrderHistoryComponent implements OnInit{
+    public modelRef: BsModalRef;
+    public modalRef: BsModalRef;
+    public  isMulticurrencyAccount : true;
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/purchase/order-history/order-history.component.ts:20]: JS doc

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.html [hunk 9]
```diff
@@ -956,1168 +1187,1389 @@
...
+                                                                </li>
+                                                                <li class="clearfix mrT1" *ngIf="isSalesInvoice">
+                                                                    <input type="text" (blur)="calculateBalanceDue()" placeholder="Amount" name="dueAmount" class="form-control amountField text-right pull-right max100" [(ngModel)]="depositAmount" [prefix]="depositCurrSymbol" [suffix]="selectedSuffixForCurrency" [mask]="inputMaskFormat"
+                                                                    />
+                                                                    <label class="pull-right pt-1">Amount</label>
+                                                                </li>
+                                                                <label class="pull-right">Adjust Amount</label>
+                                                            </ng-container>
+
+                                                            <ng-container *ngIf="!isUpdateMode">
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.html:1556]: Please check this

File: apps/web-giddh/src/app/invoice/preview/models/purchase-bill-template-preview/purchase-bill-template-preview.component.html [hunk 10]
```diff
@@ -0,0 +1,385 @@
...
+                                            <ng-container>
+                                                <tr>
+                                                    <td colspan="3">
+                                                        <h3 class="text-right title-template">
+                                                            Purchase Bill</h3>
+                                                    </td>
+                                                </tr>
+                                                <tr>
+                                                    <td>
+                                                        <span class="mb-0">Delivery</span>11/09/2017
```
[the remark this answers is not in the document]
Comment (shalinee-walkover) [reply apps/web-giddh/src/app/invoice/preview/models/purchase-bill-template-preview/purchase-bill-template-preview.component.html:43]: This is unsued file, will use it in future

**Pull request closed without merging**